go/libraries/doltcore/merge/action.go

```diff
@@ -98,26 +103,39 @@ func NewMergeSpec(ctx context.Context, rsr env.RepoStateReader, ddb *doltdb.Dolt
 		Squash:          squash,
 		Msg:             msg,
 		Noff:            noff,
+		NoCommit:        noCommit,
+		NoEdit:          noEdit,
 		Force:           force,
 		Email:           email,
 		Name:            name,
 		Date:            date,
-	}, true, nil
+	}, nil
 }
 
+// MergeCommitSpec returns whether to create new commit and MergeStats object after performing merge and commits
+// if applied. If merge can fast-forward, no commit is returned. ExecNoFFMerge function call commits after merge.
+// If merge is not fast-forward, 'no-commit' flag is not defined and if there are no conflicts and/or
+// constraint violations, commit(true) is returned.
```

Comment:
Good function docs. These will be a big help to future readers of the code! How about this small tweak to make it read better... let me know if this isn't still accurate... 

```suggestion
// MergeCommitSpec applies a merge spec, potentially fast-forwarding the current branch HEAD, and returns two 
// results: true if a new Dolt commit needs to be created to complete the merge, and a MergeStats object.
// If the merge can be applied as a fast-forward merge, no commit is needed, so false is returned for the first result.
// If the merge is a fast-forward merge, but --no-ff has been supplied, the ExecNoFFMerge function will call
// commit after merging. If the merge is not fast-forward, the 'no-commit' flag is not defined, and if there are 
// no conflicts and/or constraint violations, commit(true) is returned.
```

Actually... after rewriting this method doc comment... this seems like an odd interface. I think it's okay for this iteration, but take a step back and think about using this API as a caller. It's not intuitively obvious why this method behaves like it does – it commits in some cases, but not in others and the caller has to commit. This is a signal that we could simplify the code a bit more. Can we push the responsibility to create the commit into this method so that it's easier to explain how this method works? That could also let us remove the "doCommit" returned result, since the caller wouldn't need a signal to do a commit in some edge cases. 

Totally fine to do this as a separate PR if you prefer, so we can get this change landed in main first. Either way, I think it's a good experiment to try to see if we can simplify the interface a little bit and push more consistent responsibility into this method (which is sometimes responsible for making a commit, sometimes not).